force-app/main/default/classes/OpportunityHelper.cls

```diff
@@ -0,0 +1,203 @@
+public with sharing class OpportunityHelper extends TriggerHandler{
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id,Opportunity> newOppMap;
+    private Map<Id,Opportunity> oldOppMap;
+    
+    public OpportunityHelper() {
+        this.newOppList = (List<Opportunity>)Trigger.new;
+        this.oldOppList = (List<Opportunity>)Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>)Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>)Trigger.oldMap;
+
+    }
+
+    public override void beforeInsert(){
+        setDefaultTypeForNewOpps(newOppList);
+    }
+
+    public override void afterInsert(){
+        List<Task> insertTaskList = createTaskOnNewOpps(newOppList);
+
+        Database.insert(insertTaskList);
```

Comment:
Why are you doing a Database.insert instead of a standard insert?
It's not right or wrong just cruious